scripts/entrypoint.sh

```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -e
-
-# Command exec
-echo Entrypoint executing: "$@"
-exec "$@"
+# Copies overides
+cat build.properties.example /overrides/build.properties > build.properties  2>/dev/null || cp build.properties.example build.properties
+# Builds
+mvn clean package war:exploded -Dhttps.protocols=TLSv1.2 -Dmaven.buildNumber.doCheck=false -Dmaven.buildNumber.doUpdate=false 
+# Runs
+mvn jetty:run -Dhttps.protocols=TLSv1.2 -Dmaven.buildNumber.doCheck=false -Dmaven.buildNumber.doUpdate=false
```

Comment:
This entrypoint.sh script is shaping up nicely, handling the overrides here makes sense.

One last note: I think either the README needs a change, or this script needs a change.

With `CMD [ "/entrypoint.sh" ]` this entrypoint.sh script will be executed, as intended, but if a user follows the README and does a `docker run ... mvn clean...` they won't get the built-in overrides copy and default maven params. If they simply wanted to extend CMD and add a param or two they would have to write some huge one-liner to get the same behaviour that the container probably should be run with, the code in this file.

If that really is the intended behaviour, we should update the README to reflect that and remove the command from the `docker run` examples.

However I'm wondering if we could get the best of both worlds, allowing users to modify CMD if needed and still keeping the defacto entrypoint.sh setup that we specify in build. What do you think about this:


```dockerfile
# Dockerfile
ENTRYPOINT ["/entrypoint.sh"]
CMD mvn jetty:run -Dhttps.protocols=TLSv1.2 -Dmaven.buildNumber.doCheck=false -Dmaven.buildNumber.doUpdate=false
```

```bash
# entrypoint.sh
...
# Runs
exec "$@"
```

This would allow the container to run both with and without a custom command and would give users direct access to the maven params for easy overrides.

Is this of any use for our users? Should we lock down CMD and encourage them to use a volumed-in overrides file? Any other ideas?

Reply:
So as far as I am aware there are no overrides used when building the container. If a user tries to use the overrides with the normal container then it will fail with that entry point of `"$@"`. I have added two separate scripts that get added. The default one does what you are looking for and should all for additional expectation od the `jetty:run` there is also the overrides which allows for `build.properties` overrides.

Reply:
👍 sounds good. Thanks for the investigation.